src/VestedRewardsDistribution.t.sol

```diff
@@ -0,0 +1,545 @@
+        uint256 timeSkip = k.vestParams.tau / totalDistributions;
+
+        uint256[] memory balances = new uint256[](totalDistributions);
+        vm.warp(k.vestParams.bgn);
+        for (uint256 i = 0; i < totalDistributions; i++) {
+            skip(timeSkip);
+            k.dist.distribute();
+            balances[i] = k.rewardsToken.balanceOf(address(k.farm));
+        }
+
+        uint256[] memory deltas = new uint256[](totalDistributions - 1);
+        for (uint256 i = 1; i < totalDistributions; i++) {
+            deltas[i - 1] = balances[i] - balances[i - 1];
+        }
+
+        // Check the final balance. Allow for 0,01% error tolerance due to rounding errors.
+        uint256 tolerance = 0.0001e18;
+
+        // Check if balance grows at the same pace every time:
+        for (uint256 i = 1; i < deltas.length; i++) {
+            assertApproxEqRel(
+                deltas[i],
+                deltas[i - 1],
+                tolerance,
+                string.concat("Bad balance change between #", toString(i - 1), " and #", toString(i))
+            );
+        }
+
+        assertApproxEqRel(k.rewardsToken.balanceOf(address(k.farm)), k.vestParams.tot, tolerance);
+    }
+
+    function testDistributeLinearRampUp() public {
+        skip(l.vestParams.tau);
+
+        assertEq(l.rewardsToken.balanceOf(address(l.farm)), 0);
+
+        vm.expectEmit(false, false, false, true, address(l.dist));
+        emit Distribute(l.vestParams.tot);
+        l.dist.distribute();
+
+        // Allow for 0,01% error tolerance due to rounding errors.
+        uint256 tolerance = 0.0001e18;
+        assertApproxEqRel(l.rewardsToken.balanceOf(address(l.farm)), l.vestParams.tot, tolerance);
+    }
+
+    function testDistributeLinearRampUpEverIncreasingFuzz(uint256 totalDistributions) public {
+        // Anything between 1 per week and 1 per month.
+        totalDistributions = (bound(totalDistributions, 12, 52) * l.vestParams.tau) / 365 days;
+        // Make distributions uniformly spread across time
+        uint256 timeSkip = l.vestParams.tau / totalDistributions;
+
+        uint256[] memory balances = new uint256[](totalDistributions);
+        vm.warp(l.vestParams.bgn);
+        for (uint256 i = 0; i < totalDistributions; i++) {
+            skip(timeSkip);
+            l.dist.distribute();
+            balances[i] = l.rewardsToken.balanceOf(address(l.farm));
+        }
+
+        uint256[] memory deltas = new uint256[](totalDistributions - 1);
+        for (uint256 i = 1; i < totalDistributions; i++) {
+            deltas[i - 1] = balances[i] - balances[i - 1];
+        }
+
+        // Check if balance grows faster every time:
+        for (uint256 i = 1; i < deltas.length; i++) {
+            assertGt(
+                deltas[i],
+                deltas[i - 1],
+                string.concat("Bad balance change between #", toString(i - 1), " and #", toString(i))
+            );
+        }
```

Comment:
Can we not only check that this is increasing, but also that it is increasingly roughly linearly?

Reply:
Done.